### DISCUSSION TITLE
2023/07/11/install-mc-bds-on-android-phone-with-no-cost/

### DISCUSSION THREAD

**Discussion by giscus** [hmsjy2017/blog-comment#6] [Announcements]
# 2023/07/11/install-mc-bds-on-android-phone-with-no-cost/

前言你是否需要搭建 BDS 但买不起昂贵的服务器？本文将介绍使用 Termux 搭建 BDS 的完整步骤，只需一台安卓手机。 开始你需要准备一台安卓手机（或平板），推荐 Android 版本：7-11。 如果 Android 版本高于 11，会出现杀进程的情况（signal 9），需要连上 adb shell 手动修复。 12adb shell device_config put activity

https://blog.iamsjy.com/2023/07/11/install-mc-bds-on-android-phone-with-no-cost/

**Comment by FokersBoolean** [comment]
玩家进入时似乎会直接卡死控制台，并且卡在正在连接服务器

**Reply by fast0528** [reply]
用小窗口打开，在你进服的时候摁几次enter键就进去了😁当然，也可能会是崩溃

**Comment by fast0528** [comment]
成功实现，也能进去
不过，不建议使用那个安装脚本
那个安装脚本安装的ubuntu版本过低
使用box64时会提示GLIBC版本过低
~~尝试过安装，个人觉得过于麻烦，安装失败了~~
然后使用termux推荐的方法安装了最新版ubuntu
最新版ubuntu的默认GLIBC版本较高，直接能用了无需更新！

**Reply by hmsjy2017** [reply]
谢谢提醒，用proot-distro确实更好（我也试过）
~~主要是因为GitHub在国内访问不畅才用那个脚本的~~

**Comment by Wertester** [comment]
--2024-05-12 17:24:42--  http://unzip/               Resolving unzip (unzip)... failed: Name or service not known.                                             wget: unable to resolve host address ‘unzip’         --2024-05-12 17:24:53--  http://bds.zip/             Resolving bds.zip (bds.zip)... failed: Name or service not known.                                         wget: unable to resolve host address ‘bds.zip’       FINISHED --2024-05-12 17:24:53--                     Total wall clock time: 41s                           Downloaded: 1 files, 122M in 24s (5.13 MB/s)

不知道怎么劫掠

**Reply by Wertester** [reply]
解决

**Reply by hmsjy2017** [reply]
```
wget https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.81.01.zip -O BDS.zip
unzip BDS.zip && rm BDS.zip
```

**Reply by Wertester** [reply]
谢谢

**Comment by Wertester** [comment]
inflating: server.properties                          rm: cannot remove 'BDS.zip9与': No such file or directory                                                       root@localhost:~/mc# nano server.properties             root@localhost:~/mc# nano server.properties             root@localhost:~/mc# chmod +x bedrock_server            root@localhost:~/mc# LD_LIBRARY_PATH=. box64 ./bedrock_server                                                   box64: /lib/aarch64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by box64)                     box64: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by box64)

**Reply by fast0528** [reply]
没有文件"BDS.zip9与"无法删除，你写错名字# 了吧，然后就是GLIBC版本低了要么手动安装，要么从新安装最新版ubuntu~~(那个脚本安装的ubuntu是旧版本)~~

**Reply by hmsjy2017** [reply]
文章已更新，请使用新方法部署

**Reply by Wertester** [reply]
麻烦了

**Comment by MCyangguang** [comment]
libc++abi: terminating due to uncaught exception of type std::__1::system_error: Operation not permitted
NativeBT: ./bedrock_server() [0x34a25ebc]
NativeBT: [0x759948063c]
NativeBT: /lib/aarch64-linux-gnu/libc.so.6(+0x87628) [0x7599337628]
NativeBT: /lib/aarch64-linux-gnu/libc.so.6(gsignal+0x1c) [0x75992ecb3c]
NativeBT: /lib/aarch64-linux-gnu/libc.so.6(abort+0xf4) [0x75992d7e00]
NativeBT: [0x7523274e64]
EmulatedBT: box64(abort+0) [0x60080]
11722|SIGABRT @0x7599337628 (???(/lib/aarch64-linux-gnu/libc.so.6+0x87628)) (x64pc=0x60080/"???", rsp=0x7594c5bc68, stack=0x7594a5c000:0x7594c5c000 own=0x7594a5c000 fp=0x752c2d3be0), for accessing 0x27ae00002db5 (code=-6/prot=0), db=(nil)((nil):(nil)/(nil):(nil)/???:clean, hash:0/0) handler=0x107c6b8e0
RSP-0x20:0x000000752d951fb0 RSP-0x18:0x0000000000000000 RSP-0x10:0x0000000036ca14b0 RSP-0x08:0x0000000000000014
RSP+0x00:0x0000000102e1f83b RSP+0x08:0x0000003000000008 RSP+0x10:0x0000007594c5bd60 RSP+0x18:0x0000007594c5bc90
RAX:0x000000000000000a RCX:0x0000007594c5bc70 RDX:0x0000007599462550 RBX:0x0000007599461600 
RSP:0x0000007594c5bc68 RBP:0x000000752c2d3be0 RSI:0x0000000000000001 RDI:0x000000000000000a 
 R8:0x0000000108897dd8  R9:0x0000000000000000 R10:0x0000003f00065220 R11:0x0000007594c5b7e0 
R12:0x000000752c7d5680 R13:0x000000752d951fb0 R14:0x0000000100d2f925 R15:0x00000075843e27c0 
ES:0x002b CS:0x0033 SS:0x002b DS:0x002b FS:0x0043 GS:0x0053 
[2024-08-26 04:01:12:913 INFO] Package: com.mojang.minecraft.dedicatedserver
Version: 1.21.20.03
OS: Linux
Server start: 2024-08-26 03:57:00 UTC
Dmp timestamp: 2024-08-26 04:01:12 UTC
Upload Date: 2024-08-26 04:01:12 UTC
Session ID: 21d01407-441e-4aeb-8e0c-1f4c635c2d12
Commit hash: 71ab5eb9d016c4ec310b413fbe291f3cfea6e545
Build id: 26660154
CrashReporter Key: 04b2008f-d98c-3dd4-be42-f8ec980980e9

Crash
[2024-08-26 04:01:12:914 INFO]

**Reply by MCyangguang** [reply]
行了，果然多按几次Enter就行了

**Reply by fast0528** [reply]
额，多摁几次enter键是说进服一直加载，你这个我看了一下我的服务器日志，也出现过，应该就只是崩了而已，重开就好![Screenshot_2024-08-26-12-30-55-570_com.termux.png](https://github.com/user-attachments/assets/bf361e90-6460-42e0-a0d7-6e3bc05c29ca)

**Reply by fast0528** [reply]
不过我进服前也喜欢多摁几次enter

**Comment by LineXic** [comment]
弄好是弄好了，就是在哪里显示IP啊
![Screenshot_2024_1124_225845.jpg](https://s2.loli.net/2024/11/24/v2sNdcVrSRaLwe6.jpg)

**Reply by LineXic** [reply]
不用了，解决了

**Reply by wm1524** [reply]
在哪儿

**Reply by LineXic** [reply]
192.168.1.1端口是默认

**Comment by C3Mira** [comment]
root@localhost:~# mkdir mc && cd mc
root@localhost:~/mc# wget https://www.minecraft.net/bedrockdedicatedserver/bin-linux/bedrock-server-1.21.51.02.zip -O BDS.zip
--2025-01-11 11:27:31--  https://www.minecraft.net/bedrockdedicatedserver/bin-linux/bedrock-server-1.21.51.02.zip
Resolving www.minecraft.net (www.minecraft.net)... 198.18.0.15
Connecting to www.minecraft.net (www.minecraft.net)|198.18.0.15|:443... connected.
HTTP request sent, awaiting response... No data received.
Retrying.

--2025-01-11 11:27:47--  (try: 2)  https://www.minecraft.net/bedrockdedicatedserver/bin-linux/bedrock-server-1.21.51.02.zip
Connecting to www.minecraft.net (www.minecraft.net)|198.18.0.15|:443... connected.
HTTP request sent, awaiting response... Read error (Software caused connection abort) in headers.
Retrying.

--2025-01-11 11:28:02--  (try: 3)  https://www.minecraft.net/bedrockdedicatedserver/bin-linux/bedrock-server-1.21.51.02.zip
Connecting to www.minecraft.net (www.minecraft.net)|198.18.0.15|:443... ^C


卡在这一步了，使用代理也是这样

**Reply by fast0528** [reply]
尝试了一下，无法下载
说一个可行的吧，手机浏览器下载后在复制过去就好了，你可以下载后用mt管理器给他塞进去，可以用ubuntu到下载目录复制文件

**Reply by fast0528** [reply]
简单来个复制教程
首先你使用任意浏览器下载文件，记录好你下载的文件名以及位置
![IMG_20250111_201644.jpg](https://github.com/user-attachments/assets/ca6b4ea1-d6c7-4098-a299-0b01aca4a4db)

下载完成后，我是存储在/storage/emulated/0/Download目录下的
![Screenshot_2025-01-11-20-14-41-348_bin.mt.plus-edit.jpg](https://github.com/user-attachments/assets/bbdf8eaf-fd97-4c6c-8ac2-4eea57c597cb)

然后使用
```
cp /storage/emulated/0/Download/bedrock-server-1.21.51.02.zip ./
```

将bedrock-server-1.21.51.02.zip文件复制到当前目录即可，也可使用mv命令
![Screenshot_2025-01-11-20-18-50-642_com.termux-edit.jpg](https://github.com/user-attachments/assets/74d26bef-379a-412a-9d44-d0e1a1496834)

**Comment by yucho123987** [comment]
测试发现通过box64运行1.19.50.02版本的bds会自动退出，输出如下：
```
root@localhost:~/bds# LD_LIBRARY_PATH=. box64 ./bedrock_server                                      [BOX64] Box64 arm64 v0.3.9 d0045b15 with Dynarec built on Oct 16 2025 06:43:52                      [BOX64] Dynarec for ARM64, with extension: ASIMD AES CRC32 PMULL ATOMICS SHA1 SHA2                  [BOX64] Running on Cortex-A55 with 8 cores, pagesize: 4096                                          [BOX64] Will use hardware counter measured at 13.0 MHz emulating 3.3 GHz                            [BOX64] Didn't detect 48bits of address space, considering it's 39bits                              [BOX64] Warning, program break not found          [BOX64] Counted 24 Env var                        [BOX64] Library search path:                      [BOX64] Binary search path: ./:bin/:/usr/local/sbin/:/usr/local/bin/:/usr/sbin/:/usr/bin/:/sbin/:/bin/:/usr/local/games/:/usr/games/:/data/data/com.termux/files/usr/bin/:/system/bin/:/system/xbin/    [BOX64] Looking for ./bedrock_server              [BOX64] Warning: cannot create memory map (@0x200000 0x6c348b0) for elf "/root/bds/bedrock_server"[BOX64]  got 0x711661b000                           [BOX64] Rename process to "bedrock_server"        [BOX64] Using native(wrapped) libdl.so.2          [BOX64] Using native(wrapped) libcurl.so.4        [BOX64] Using emulated /usr/lib/box64-x86_64-linux-gnu/libstdc++.so.6                               [BOX64] Using native(wrapped) libm.so.6           [BOX64] Using emulated /usr/lib/box64-x86_64-linux-gnu/libgcc_s.so.1                                [BOX64] Using native(wrapped) libpthread.so.0     [BOX64] Using native(wrapped) libc.so.6           [BOX64] Using native(wrapped) ld-linux-x86-64.so.2[BOX64] Using native(wrapped) libutil.so.1        [BOX64] Using native(wrapped) librt.so.1          [BOX64] Using native(wrapped) libbsd.so.0         [BOX64] 26254|Warning, calling Signal 11 function handler SIG_DFL                                   [BOX64] Unhandled signal caught, aborting         [BOX64] NativeBT: /usr/local/bin/box64() [0x34b00f68]                                               [BOX64] NativeBT: /usr/local/bin/box64() [0x34b416e8]                                               [BOX64] NativeBT: [0x711d63d84c]                  [BOX64] NativeBT: /lib/aarch64-linux-gnu/libc.so.6(+0x87dc0) [0x711d4f7dc0]                         [BOX64] NativeBT: /lib/aarch64-linux-gnu/libc.so.6(gsignal+0x20) [0x711d4a6980]                     [BOX64] NativeBT: /lib/aarch64-linux-gnu/libc.so.6(abort+0x2c) [0x711d491ac4]                       [BOX64] NativeBT: /usr/local/bin/box64() [0x34b3ebe8]                                               [BOX64] NativeBT: /usr/local/bin/box64() [0x34b407c0]                                               [BOX64] NativeBT: /usr/local/bin/box64() [0x34b41100]                                               [BOX64] NativeBT: /usr/local/bin/box64() [0x34b0115c]                                               [BOX64] NativeBT: /usr/local/bin/box64() [0x34b015d8]                                               [BOX64] NativeBT: /usr/local/bin/box64() [0x35388780]                                               [BOX64] NativeBT: /usr/local/bin/box64() [0x34b16b9c]                                               [BOX64] NativeBT: /usr/local/bin/box64(my___libc_start_main+0x108) [0x34b24188]                     [BOX64] NativeBT: /usr/local/bin/box64() [0x34b87980]                                               [BOX64] NativeBT: [0x711429ff58]                  [BOX64] EmulatedBT: ??? [0x1f670e0]               [BOX64] 26254|SIGABRT @0x711d4f7dc0 (???(/lib/aarch64-linux-gnu/libc.so.6+0x87dc0)) (x64pc=0x1f670e0/"???", rsp=0x71166194a8, stack=0x7115e1b000:0x711661b000 own=(nil) fp=0x205), for accessing 0x28100000668e (code=-6/prot=0), db=(nil)((nil):(nil)/(nil):(nil)/???:clean, hash:0/0) handler=(nil)       RSP-0x20:0x0000000000000000 RSP-0x18:0x0000000000000000 RSP-0x10:0x0000000000000000 RSP-0x08:0x0000000000000000                                       RSP+0x00:0x0000000000000000 RSP+0x08:0x0000000000000000 RSP+0x10:0x0000000000000000 RSP+0x18:0x0000000000000002                                       RAX:0x0000000000000000 RCX:0x000000711d12dda0 RDX:0x0000007116619eb8 RBX:0x0000000000000000         RSP:0x00000071166194a8 RBP:0x0000000000000205 RSI:0x0000007116619ea8 RDI:0x0000000000000001          R8:0x000000711d12de10  R9:0x0000000030000080 R10:0x000000000000000f R11:0x0000000000000006         R12:0x000000711d138008 R13:0x0000000000000001 R14:0x0000007116619ea8 R15:0x0000007116619eb8         ES:0x002b CS:0x0033 SS:0x002b DS:0x002b FS:0x0043 GS:0x0053                                         Aborted
```
但是最新版本的bds却能够正常运行，不知道是什么原因。

**Comment by wm1524** [comment]
IP在哪儿

**Reply by fast0528** [reply]
ip？手机ip呗，但是国内没ipv4的公网ip。
如果局域网的话就是路由器分配的ip。
如果你要远程联机可以找远程组网软件，或者frp软件，手机电脑都一样。毕竟…公网ip不好搞还麻烦，何况是手机的公网ip( 应该不支持获取吧 )
